File: leetcode/Average of Levels in Binary Tree/2.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func averageOfLevels(root *TreeNode) []float64 {
	list := []*TreeNode{root}
	ret := []float64{}

	for len(list) > 0 {
		size := len(list)
		total := 0
		count := 0

		for i := 0; i < size; i++ {
			curr := list[0]

			list = list[1:]
			total += curr.Val
			count++
			if curr.Left != nil {
				list = append(list, curr.Left)
			}
			if curr.Right != nil {
				list = append(list, curr.Right)
			}
		}

		ret = append(ret, float64(total)/float64(count))
	}

	return ret
}

File: leetcode/All Elements in Two Binary Search Trees/2.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    private var list1: MutableList<Int> = mutableListOf()
    private var list2: MutableList<Int> = mutableListOf()

    fun getAllElements(root1: TreeNode?, root2: TreeNode?): List<Int> {
        list1 = mutableListOf()
        list2 = mutableListOf()

        inorder(root1, list1)
        inorder(root2, list2)

        val ans: MutableList<Int> = mutableListOf()

        var idx1 = 0
        var idx2 = 0
        while (idx1 < list1.size && idx2 < list2.size) {
            if (list1[idx1] < list2[idx2]) {
                ans.add(list1[idx1])
                idx1++
            } else {
                ans.add(list2[idx2])
                idx2++
            }
        }
        while (idx1 < list1.size) {
            ans.add(list1[idx1])
            idx1++
        }
        while (idx2 < list2.size) {
            ans.add(list2[idx2])
            idx2++
        }
        return ans
    }

    private fun inorder(root: TreeNode?, list: MutableList<Int>) {
        if (root == null) return
        inorder(root.left, list)
        list.add(root.`val`)
        inorder(root.right, list)
    }
}